Added Crypto utility files from Avalon Github Repo

divyataori opened the pull request:
Added crypto utility files from Avalon Github Repo.
Repo: https://github.com/hyperledger/avalon.git
folder path:
1) avalon/common/crypto_utils
modified: No

manojgop commented on crypto_utils/tests/run_test.sh:
```diff
@@ -0,0 +1,77 @@
+#!/bin/bash
```
Is the test directory and the test files still valid ? If not, please remove it from this PR. Lot of directories referred here are not present in this repo

manojgop commented:
Why is there 24 commits in this PR ? Please add just one commit. You can mention author and committer as different and mention in commit message that this file was picked from Avalon repo

dcmiddle commented:
> Why is there 24 commits in this PR ? Please add just one commit. You can mention author and committer as different and mention in commit message that this file was picked from Avalon repo

+1 to only moving relevant files.
However I think it is best practice to bring over the commit history with the files. The commit messages preserves the "why" information in the codebase.

manojgop commented:
> > Why is there 24 commits in this PR ? Please add just one commit. You can mention author and committer as different and mention in commit message that this file was picked from Avalon repo
> 
> +1 to only moving relevant files.
> However I think it is best practice to bring over the commit history with the files. The commit messages preserves the "why" information in the codebase.

Agree with preserving commit history in general.  But there are couple of issues we need to address (mentioned below). One option would be to cherry pick required commits. But it's not straightforward be fix these.  Other option is we can update the commit message in this PR mentioning that these files are picked from Avalon repo.

1) Some commit history and "commit messages" seem to be incomplete and not relevant for some files in this PR. For example, the original commit for crypto files (eg: worker_encryption.py/worker_hash.py/worker_sign.py) is missing. These files were added in the PR for Graphene integration -> https://github.com/hyperledger/avalon/pull/612/files.

2) Commit history for some files are not relevant. eg: There are commit history for files avalon_crypto_utils/signature.py which is not in this PR and not required.

3) Some commit history like "Add ias verification to generic client " is not relevant too.

manojgop committed:
Added crypto utility files from Github Avalon Main Repo.
Repo: https://github.com/hyperledger/avalon.git
Folder path: avalon/common/crypto_utils/avalon_crypto_utils
Modified : No

Signed-off-by: Divya Taori <divya.taori@wipro.com>
Co-authored-by: manju956 <manjunath.a.c@intel.com>
Co-authored-by: Ramakrishna Srinivasamurthy <ramakrishna.srinivasamurthy@intel.com>
Co-authored-by: Divya Taori <divya.taori@wipro.com>

divyataori commented:
Updated. Please review.

rranjan3 commented on src/common/crypto_utils/avalon_crypto_utils/worker_signing.py:
```diff
@@ -0,0 +1,294 @@
+        """
+        Utils function to verify integrity of worker encryption key using
+        worker verification key
+        @params encryption_key_signature - Signature computed on hash
```
Since this a new file being introduced, ensure the function description style is consistent.

divyataori replied:
These files are from the main Avalon repo and have not been modified. Will match the function description style in a different PR.
